Libs/vtkSegmentationCore/vtkSegmentation.cxx

```diff
@@ -1573,13 +1573,24 @@ void vtkSegmentation::SeparateSegmentLabelmap(std::string segmentId)
     thresholdErase->SetInValue(0);
     thresholdErase->ReplaceOutOff();
     thresholdErase->Update();
+
+    // Although the source labelmap will be modified, onSourceRepresentationModified would invalidate the non-source representations.
```

Comment:
```suggestion
    // Although the source labelmap will be modified, onSourceRepresentationModified would unnecessarily invalidate the non-source representations
    // (because separating a labelmap into a new layer does not change the segment's shape).
```